### PR TITLE
test: add property based tests to DataBus

### PR DESCRIPTION
**Pull request by ziqiangai**
Submit a pull request for this project.

# Why? 

For the issue #140 
Use `fast-check.js` to simply test the two commands of DataBus to test whether the test quality of DataBus can be improved.

# What?

I modified the `SaveOps` logic of `MockDataLoadersaver`.

At the same time, a test script is added. If the test code is valid, we can expand the test of `DataBus` on the basis of this.

### CODE REVIEW

File: packages/core/src/databus/__tests__/databus-fast-check.test.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        if (result.result === ExecuteResult.Success) {
+          expectRevisionChangedCount += 1;
+        }
+
+        // the id, name, and type field of the Datasheet are not changed.
+        const prosEq = (dst1!.id === 'dst1' && dst1!.type === ResourceType.Datasheet && dst1!.name === 'datasheet 1');
+        expect(dst1!.id).toStrictEqual('dst1');
+        expect(dst1!.type).toStrictEqual(ResourceType.Datasheet);
+        expect(dst1!.name).toStrictEqual('datasheet 1');
+        return new Promise(resolve => {
```
Comment (arucil) [inline packages/core/src/databus/__tests__/databus-fast-check.test.ts:63]: A simple `return prosEq` is enough.

File: packages/core/src/databus/__tests__/databus-fast-check.test.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        if (result.result === ExecuteResult.Success) {
+          expectFieldAddedCount += 1;
+        }else {
+          console.error('some error happens');
+        }
+
+        const fieldCount = Object.keys(dst1.fields).length;
+        return new Promise(resolve => {
+          resolve(fieldCount >= (oldFieldCount + 1));
+        });
```
Comment (arucil) [inline packages/core/src/databus/__tests__/databus-fast-check.test.ts:115]: missing assertions inside this function

**Review by arucil — CHANGES_REQUESTED** [review]

**Comment by arucil** [comment]
Great job. Only some minor problems remain, fix them then everything is good.

**Comment by ziqiangai** [comment]
This pr was accidentally turned off by me, the new pr is located at [Pull Request](https://github.com/apitable/apitable/pull/467)

**Pull request closed without merging**